x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/crud/rules_management_client.ts

```diff
@@ -0,0 +1,379 @@
+  ) => Promise<RuleAlertType>;
+  importRule: (importRulePayload: ImportRuleProps) => Promise<RuleAlertType>;
+}
+
+export const createRulesManagementClient = (rulesClient: RulesClient) => {
+  const client = {
+    createCustomRule: async (
+      createCustomRulePayload: CreateCustomRuleProps
+    ): Promise<RuleAlertType> => {
+      return createCustomRule(rulesClient, createCustomRulePayload);
+    },
+
+    createPrebuiltRule: async (
+      createPrebuiltRulePayload: CreatePrebuiltRuleProps
+    ): Promise<RuleAlertType> => {
+      return createPrebuiltRule(rulesClient, createPrebuiltRulePayload);
+    },
+
+    updateRule: async (updateRulePayload: UpdateRuleProps): Promise<RuleAlertType> => {
+      return updateRule(rulesClient, updateRulePayload);
+    },
+
+    patchRule: async (patchRulePayload: PatchRuleProps): Promise<RuleAlertType> => {
+      return patchRule(rulesClient, patchRulePayload);
+    },
+
+    deleteRule: async (deleteRulePayload: DeleteRuleProps): Promise<void> => {
+      return deleteRule(rulesClient, deleteRulePayload);
+    },
+
+    upgradePrebuiltRule: async (
+      upgradePrebuiltRulePayload: UpgradePrebuiltRuleProps
+    ): Promise<RuleAlertType> => {
+      return upgradePrebuiltRule(rulesClient, upgradePrebuiltRulePayload);
+    },
+
+    importRule: async (importRulePayload: ImportRuleProps): Promise<RuleAlertType> => {
+      return importRule(rulesClient, importRulePayload);
+    },
+  };
+
+  return client;
+};
+
+export const createCustomRule = async (
+  rulesClient: RulesClient,
+  createCustomRulePayload: CreateCustomRuleProps
+): Promise<RuleAlertType> => {
+  const { params } = createCustomRulePayload;
+
+  const rule = await _createRule(rulesClient, params, { immutable: false });
+  return rule;
+};
+
+export const createPrebuiltRule = async (
+  rulesClient: RulesClient,
+  createPrebuiltRulePayload: CreatePrebuiltRuleProps
+): Promise<RuleAlertType> => {
+  const { ruleAsset } = createPrebuiltRulePayload;
+
+  const rule = await _createRule(rulesClient, ruleAsset, {
+    immutable: true,
+    defaultEnabled: false,
+  });
+
+  return rule;
+};
+
+export const updateRule = async (
+  rulesClient: RulesClient,
+  updateRulePayload: UpdateRuleProps
+): Promise<RuleAlertType> => {
+  const { ruleUpdate, existingRule } = updateRulePayload;
+
+  const update = await _updateRule(rulesClient, { ruleUpdate, existingRule });
+
+  const enabled = ruleUpdate.enabled ?? existingRule.enabled;
+  await _toggleRuleEnabledOnUpdate(rulesClient, existingRule, enabled);
+
+  return { ...update, enabled };
+};
+
+export const patchRule = async (
+  rulesClient: RulesClient,
+  patchRulePayload: PatchRuleProps
+): Promise<RuleAlertType> => {
+  const { nextParams, existingRule } = patchRulePayload;
+  const update = await _patchRule(rulesClient, patchRulePayload);
+
+  await _toggleRuleEnabledOnUpdate(rulesClient, existingRule, nextParams.enabled ?? false);
```

Comment:
I'm not sure I understand this `nextParams.enabled ?? false` part. If `enabled` hasn't been passed with `nextParams`, the rule will be disabled, which doesn't seem correct.

Reply:
True, thanks. Fixed 👍